src/span.js

```diff
@@ -138,13 +138,15 @@ export default class Span {
 
     /**
      * Checks whether or not a span can be written to.
+     * A span can be written to if it:
+     *  - 'finish' has not been called on the span.
+     *  - the span is not finalized.
+     *  - the span has been finalized, but also is sampled.
+     *  - the span has not been serialized using injectors.
```

Comment:
unnecessary complex. Break into two parts:
  * span is finalized when either of these happens:
    * it had inherited sampling decision from its parent
    * it is finish()-ed
    * setOperationName is called
    * it it used as a parent for another span
    * its context is serialized using injectors

Then isWritable is simply `!finalized || sampled`